tests/virtualization/livemigration/live_migration_dst.pm

```diff
@@ -0,0 +1,62 @@
+# Copyright © 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Live migration test for destination host
+#
+# - Wait for guest migration initiated from source host after installation.
+#
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package live_migration_dst;
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use Utils::Systemd 'systemctl';
+use lockapi;
+use mmapi;
+use virt_autotest::utils 'collect_virt_system_logs';
+require 'sle/tests/virtualization/livemigration/live_migration_src.pm';
+
+my $images_path = "/var/lib/libvirt/images";
+
+barrier_create('let_dst_upload_logs', 2);
```

Comment:
@tonyyuan1 Can you explain a little bit the usage of barrier_create/barrier_wait? Any difference with mutex_create/mutex_wait? I tried to read the api in os-autoinst code, but no description either. So I am not sure the logic around this let_dst_upload_logs lock.

Reply:
When the test fails in child job of two machines parallel mm-test, src worker just kills the backend after collecting logs and uploading. The dst worker cancels the parent job when it finds the backend killed. The post_fail_hook and log collection never happens in parent job if the mm-test failure triggered by child test failure.

barrier_create can solve this problem. 
barrier_create('let_dst_upload_logs', 2) creates two barriers. They must be released no matter if the test fail or succeed. 
You can take "barrier_wait" as barrier releasing.
When the code of the child and parent both arrive at the "barrier_wait" they can proceed. Otherwise, one has to wait for the other. 
The code on dst side is simple so it reaches to barrier_wait earlier than src code.  When the src test fails it jumps to post_fail_hook  to release the second barrier.  Both src and dst start to collect logs. 

 When the src test succeeds it jump to post_run_hook which creates test_done file on shared storage. dst can access the shared storage too. It just releases its barrier and skip log collection.

Reply:
Got it, thanks for the explanations!

Reply:
Ok, will add it.